### ISSUE TITLE
obs_nodding and obs_nodding_irregular do not seem to behave well with multiple exposures per nod

### ISSUE THREAD

**Issue by Hoeijmakers** [tomasstolker/pycrires#11]
I have been trying to reduce an ABBA sequence where each A and B nod consists of 5 exposures. So the sequence goes:
A,A,A,A,A,B,B,B,B,B, B,B,B,B,B,A,A,A,A,A, etc.

When using `pipeline.obs_nodding`, it takes exposure #5 (A) and exposure #6 (B) as a nodding pair. Exposures #1 to #4 and #7-10 are ignored entirely.

When using `pipeline.obs_nodding_irregular`, it starts by taking exposure #1 (A) and exposure #6 (B), which I think is the desired behaviour, and then continues with exposure #2 (A) and exposure #6 (B) again, and the remaining B exposures of the A-B pair are subsequently ignored.

I believe that a solution should ideally be at the level of `obs_nodding` (because this sequence is regular), in that it recognises the number of A and B frames in each nod pair correctly.

**Comment by Hoeijmakers** [comment]
I agree that merging frames is probably the best, most generic way to deal with this. That may be a bit easier said than done, if all those frames need to be individually pipeline reduced before being co-added. The most logical place where this should ultimately be addressed is probably the cr2res_obs_nodding pipeline recipe itself.

For now I have created a PR with a temporary fix based on obs_nodding_irregular, adding an optional keyword to pair A's and B's one to one in sequence.

**Comment by tomasstolker** [comment]
That's true, although `util_calib` can be used with `calib_type='nodding'` for calibrating the science frames. The output could then be collapsed by subsets and used for the pair-wise subtraction with `obs_nodding`. But this may not be something that can be quickly implemented indeed.

Thanks a lot for creating the PR with a temporary fix! That seem indeed a good solution 👍.
